### PR TITLE
Update Minimum Living Wage age

### PR DESCRIPTION
**Pull request by AgaDufrat**
Since 2021-04-01 minimum living wage age changed from 25 to 23 but some parts of the calculator were never updated.

This caused to incorrectly display the outcome as "National Minimum Wage" instead of "National Living Wage" for those age 23 and 24. For example: https://www.gov.uk/am-i-getting-minimum-wage/y/current_payment/apprentice_over_19_second_year_onwards/23/7/40.0/480.0/no/no/no

It also removes text_for :error_valid_age_for_living_wage? block as we never validate it in the flow.

### How to test it

- for age below 23 "You appear to be getting the National Minimum Wage."
/am-i-getting-minimum-wage/y/current_payment/apprentice_over_19_second_year_onwards/22/7/40.0/380.0/no/no/no

- for age 23 and above "You appear to be getting the National Living Wage."
/am-i-getting-minimum-wage/y/current_payment/apprentice_over_19_second_year_onwards/23/7/40.0/380.0/no/no/no

https://trello.com/c/RoCl0SSO/875-draft-fix-national-living-wage-age-in-calculators

---
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: app/flows/am_i_getting_minimum_wage_flow/questions/how_old_are_you.erb [hunk 1]
```diff
@@ -9,7 +9,3 @@
 <% text_for :error_message do %>
   Please enter a whole number greater than 0
 <% end %>
-
```
Comment (chao-xian) [inline app/flows/am_i_getting_minimum_wage_flow/questions/how_old_are_you.erb:12]: Very minor feedback and not a blocker to merging: I would have split this out into a separate commit to be certain it doesn't affect the rest of the changes which feel unrelated.

**Review by chao-xian — APPROVED** [review]

File: test/unit/calculators/minimum_wage_calculator_test.rb [hunk 2]
```diff
@@ -90,8 +90,8 @@ class MinimumWageCalculatorTest < ActiveSupport::TestCase
       end
 
       context "for age for living wage" do
-        should "not accept ages below 25" do
-          assert_not @calculator.valid_age_for_living_wage?(24)
+        should "not accept ages below 23" do
+          assert_not @calculator.valid_age_for_living_wage?(22)
         end
 
         should "accept ages 25 or above" do
```
Comment (hannako) [inline test/unit/calculators/minimum_wage_calculator_test.rb:97]: I know this test passes, but I think it should be updated to `accept ages 23 or above`

File: test/unit/calculators/minimum_wage_calculator_test.rb [hunk 3]
```diff
@@ -113,8 +113,17 @@ class MinimumWageCalculatorTest < ActiveSupport::TestCase
         end
       end
 
-      should "return false if age is lower than 24 or nil" do
+      should "return true if the age is 23 or over, and the date is on or after 2021-04-01" do
```
Comment (hannako) [inline test/unit/calculators/minimum_wage_calculator_test.rb:116]: Why do we have to support people checking that they're eligible for NLW on dates in the past?

Comment (hannako) [reply test/unit/calculators/minimum_wage_calculator_test.rb:116]: Ah, because you can check last years eligibility can't you. https://www.gov.uk/am-i-getting-minimum-wage/y GOTTIT!

File: test/unit/calculators/minimum_wage_calculator_test.rb [hunk 4]
```diff
@@ -90,13 +90,26 @@ class MinimumWageCalculatorTest < ActiveSupport::TestCase
...
+          setup { @calculator.date = Date.parse("2020-04-01") }
+
+          should "not accept ages below 25" do
+            assert_not @calculator.valid_age_for_living_wage?(23)
+          end
+
+          should "accept ages 25 or above" do
+            assert @calculator.valid_age_for_living_wage?(25)
+            assert @calculator.valid_age_for_living_wage?(26)
+          end
```
Comment (hannako) [inline test/unit/calculators/minimum_wage_calculator_test.rb:112]: Nice 👍

**Review by hannako — APPROVED** [review]
